modules/dnn/src/layers/recurrent_layers.cpp

```diff
@@ -173,9 +173,14 @@ class LSTMLayerImpl CV_FINAL : public LSTMLayer
             CV_CheckEQ(Wh.rows, Wx.rows, "");
             CV_CheckEQ(Wh.rows, (1 + static_cast<int>(bidirectional))*4*Wh.cols, "");
             CV_CheckEQ(Wh.rows, (int)bias.total(), "");
-            CV_CheckEQ(hInternal.cols, Wh.cols, "");
-            CV_CheckEQ(hInternal.cols, cInternal.cols, "");
-            CV_CheckEQ(hInternal.rows, cInternal.rows, "");
+            // Only perform these checks if hInternal and cInternal are not empty matrices
+            // e.g. inputs are not given by a user
+            if (!hInternal.empty() && !cInternal.empty())
```

Comment:
Here and everywhere else: h and c aren't tied together - one could pass h as input and c as const blob and vice versa. They should be handled separately.

Reply:
Done! 

*May be such input behaviour should be restricted completely?